src/Compilers/Core/Portable/Syntax/SyntaxTriviaListBuilder.cs

```diff
@@ -49,7 +49,7 @@ public void Clear()
         {
             get
             {
-                if (index < 0 || index > _count)
+                if (index < 0 || index >= _count)
```

Comment:
Do we even need this bounds check at all? The index access is going to throw an `IndexOutOfRangeException` for any invalid access anyway, so this is just moving the exception one level higher in the stack.

Reply:
That makes sense, thank.